octopusdeploy_framework/resource_tenant_common_variable.go

```diff
@@ -245,6 +251,24 @@ func checkIfCommonVariableIsSensitive(tenantVariables *variables.TenantVariables
 	return false, fmt.Errorf("unable to find template for tenant variable")
 }
 
+func checkIfCandidateVariableRequiredForTenant(tenant *tenants.Tenant, tenantVariables *variables.TenantVariables, plan tenantCommonVariableResourceModel) (bool, error) {
+	if tenant.ProjectEnvironments == nil || len(tenant.ProjectEnvironments) == 0 {
+		return false, fmt.Errorf("tenant not connected to any projects")
+	}
+
+	if libraryVariable, ok := tenantVariables.LibraryVariables[plan.LibraryVariableSetID.ValueString()]; ok {
+		for _, template := range libraryVariable.Templates {
+			if template.GetID() == plan.TemplateID.ValueString() {
+				return true, nil
+			}
+		}
+	} else {
+		return false, fmt.Errorf("tenant not connected to a project that includes variable set " + plan.LibraryVariableSetID.ValueString())
```

Comment:
Nit: `fmt.Errorf` is normally only used for error strings that are formatted, its fundamentally the same as `errors.New(fmt.Sprintf("my err %s", someString))`. String concatenate with `+` is technically faster most of the time but we lose that gain by putting it in `fmt.Errorf`. We should use `errors.New` here or actually use a formatted string.

```
fmt.Errorf("tenant not connected to a project that includes variable set  %s",  plan.LibraryVariableSetID.ValueString())
```
or
```
errors.New("tenant not connected to a project that includes variable set " + plan.LibraryVariableSetID.ValueString())
```

Reply:
Went with proper format-string usage of existing `fmt.Errorf` for consistency with all the other usages in the same area, rather than introducing the `errors` import and having it feel inconsistent.